spec/actions_spec.rb

```diff
@@ -0,0 +1,179 @@
+    it '#update!' do
+      instance = klass.create!(value: 3)
+      expect(instance.value).to eq 3
+
+      expect {
+        instance.value = '4'
+      }.to raise_error(TypeError)
+
+      expect {
+        instance.update!(value: '4')
+      }.to raise_error(TypeError)
+
+      instance.update!(value: 4)
+      expect(instance.value).to eq 4
+
+      another_instance = klass.find(instance.id)
+      expect(instance.value).to eq another_instance.value
+
+      instance = klass.new(value: 3)
+      instance.update!(value: 4)
+      another_instance = klass.find(instance.id)
+      expect(instance.value).to eq another_instance.value
+    end
+
+    it '#save!' do
+      instance = klass.new(value: 3)
+      expect(instance.value).to eq 3
+      expect(instance.updated_at).to be_nil
+      expect(instance.id).to be_nil
+      expect {
+        # instance_key is not available when id is nil
+        instance.instance_key
+      }.to raise_error(TypeError)
+      instance.save!
+
+      expect(instance.created_at).not_to be_nil
+      expect(instance.updated_at).not_to be_nil
+
+      another_instance = klass.find(instance.id)
+      expect(instance.value).to eq another_instance.value
+    end
+
+    it 'doesn\'t update redis until save!/update! is called' do
+      instance = klass.create!(value: 3)
+      instance.value = 4
+
+      another_instance = klass.find(instance.id)
+      expect(another_instance.value).to eq 3
+
+      instance.save!
+      another_instance = klass.find(instance.id)
+      expect(another_instance.value).to eq instance.value
+    end
+
+    it 'resets ttl' do
+      instance = klass.create!(value: 3)
+
+      klass.ttl(2.days)
+      migrator.change_ttl_passive(klass)
+      expect(klass.redis.ttl(instance.instance_key)).to eq(-1)
+
+      instance.save!
+      expect(klass.redis.ttl(instance.instance_key) > 0).to be true
+
+      klass.ttl(nil)
+      migrator.change_ttl_passive(klass)
+      instance.update!(value: 4)
+      expect(klass.redis.ttl(instance.instance_key)).to eq(-1)
+    end
+  end
+
+  context 'read' do
+    it '#find' do
+      non_existing_id = 1
+
+      expect {
+        klass.find(non_existing_id)
+      }.to raise_error(Redcord::RecordNotFound)
+
+      instance = klass.create!(value: 1)
+
+      another_instance = klass.find(instance.id)
+
+      # It validates types
+      expect {
+        klass.find(another_instance.id.to_s)
+      }.to raise_error(TypeError)
+
+      expect(another_instance.id).to eq instance.id
+      expect(another_instance.value).to eq instance.value
+    end
+  end
+
+  context 'delete' do
+    it '#destroy' do
+      instance = klass.create!(value: 1)
+
+      another_instance = klass.find(instance.id)
+      expect(another_instance.id).to eq instance.id
+      expect(another_instance.value).to eq instance.value
+      expect(another_instance.created_at.to_i).to eq instance.created_at.to_i
+      expect(another_instance.updated_at.to_i).to eq instance.updated_at.to_i
```

Comment:
I feel like these checks should be done in most other tests as well. Maybe we can have a helper method, e.g. `eq_redcord` which checks that all ids and attributes are equal between two redcords.

Reply:
Good idea!

Reply:
This can probably come in a separate change if you want, so approving for now